src/wattsi.pas

```diff
@@ -494,7 +494,23 @@    TCrossReferences = record
                LastSeenHeadingID := MungeTopicToID(Element.TextContent.AsString);
                if (LastSeenHeadingID = '') then
                   LastSeenHeadingID := 'blank-heading';
-               LastSeenHeadingID := EnsureID(Element, LastSeenHeadingID).AsString;
+               ExtractedData := EnsureID(Element, LastSeenHeadingID);
+               LastSeenHeadingID := ExtractedData.AsString;
+
+               // Append a self-link to each header
+               if (ClassName <> 'no-num no-toc') then
+               begin
+                  HeadingSelfLink := ConstructHTMLElement(eA);
+                  HeadingSelfLink.SetAttribute('class', 'self-link');
+
+                  Scratch := Default(Rope);
+                  Scratch.Append('#');
+                  Scratch.AppendDestructively(ExtractedData);
+                  HeadingSelfLink.SetAttributeDestructively('href', Scratch);
+
+                  Element.AppendChild(HeadingSelfLink);
+               end;
+
```

Comment:
But it can be assigned again, right? You did not add such comments for other AppendDestructively. Should we comment all of them?